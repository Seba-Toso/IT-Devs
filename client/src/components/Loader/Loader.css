.lds-hourglass {
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
}
.lds-hourglass:after {
  content: " ";
  display: block;
  border-radius: 50%;
  width: 0;
  height: 0;
  margin: 8px;
  box-sizing: border-box;
  border: 32px solid #fff;
  border-color: #fff transparent #fff transparent;
  animation: lds-hourglass 1.2s infinite;
}
@keyframes lds-hourglass {
  0% {
    transform: rotate(0);
    animation-timing-function: cubic-bezier(0.55, 0.055, 0.675, 0.19);
  }
  50% {
    transform: rotate(900deg);
    animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
  }
  100% {
    transform: rotate(1800deg);
  }
}

.loader {
  background: rgb(63, 114, 190);
  width: 100%;
  height: 100vh;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 10000;
  clip-path: circle(150% at 100% 0);
  transition: clip-path 6.00s ease-in-out ;
}
.loader2 {
  clip-path: circle(0.0% at 100% 100%);
}
.lds-hourglass h1 {
color: #fff;
font-size: 30PX;
font: oblique bold 120% cursive;
}