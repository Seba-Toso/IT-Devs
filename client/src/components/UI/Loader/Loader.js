import React from 'react';
import './Loader.css';




const Loader = () => {
    window.addEventListener("load", function () {
        this.document.getElementById("loader").classList.toggle("loader2")
    })
  
    return (
        <div>
            <div class="lds-hourglass loader" id="loader">
                <h1>PREPARANDO TODO PARA TU MEJOR EXPERIENCIA...</h1>
            </div>
        </div>

    );
}

export default Loader;

