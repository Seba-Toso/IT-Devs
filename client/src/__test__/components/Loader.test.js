import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Loader from '../../components/Loader/Loader';

describe('<Loader />', () => {
  test('Loader: Must have welcome message to load.', () => {
    const loader = mount(<Loader />);
    expect(loader.find('h1')).not.toBe(null);
  });
});

describe('<Loader /> Snapshot', () => {
  test('Double check on Loader UI', () => {
    const loader = create(<Loader />);
    expect(loader.toJSON()).toMatchSnapshot();
  })
})
