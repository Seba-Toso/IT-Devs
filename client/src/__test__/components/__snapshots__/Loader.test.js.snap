// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Snapshot Double check on Loader UI 1`] = `
<div>
  <div
    className="lds-hourglass loader"
    id="loader"
  >
    <h1>
      PREPARANDO TODO PARA TU MEJOR EXPERIENCIA...
    </h1>
  </div>
</div>
`;
